### PR TITLE
Supports setting comma separated env vars

### PR DESCRIPTION
**Pull request by navidshaikh**
/lint

## Proposed Changes
 `--env ENV1=VAL1 --env ENV2=VAL2,ENV3=VAL3`

 should work specifying by separate `--env` or specifying comma separated
 values to `--env` or combination of both.

**Release Note**

```release-note
Supports specifying comma separated environment variables as `--env KEY1=VAL1,KEY2=VAL2`
```

### CODE REVIEW

**Comment by rhuss** [comment]
Fine with me, but please note that this might conflict with #342 which also has some minor tweaks on env vars. Maybe we can hold off with this one until #342 is merged ? (its soonish)

**Comment by rhuss** [comment]
@navidshaikh don't forget to add to CHANGELOG.adoc, too

**Comment by navidshaikh** [comment]
>  this might conflict with #342

@rhuss : I thought about that, separating the values by comma for `--env` should be handled by cobra and we receive the values in list before we interpret the values as mentioned #342. 

Without comma
```
--env KEY=value sets EnvVar{Name: "KEY", Value: "value"} (current behavior)
--env KEY= sets EnvVar{Name: "KEY"} (current behavior)
--env KEY
--env KEY- removes the env var matching EnvVar{Name: "KEY", ...}
```

With comma
```
--env KEY=value,KEY2=value2
--env KEY=,KEY2=
--env KEY,KEY2
--env KEY-,KEY2-
```
I think this should just work fine without any additional processing.

**Comment by duglin** [comment]
Not strong opinion either way, but do people have a guess as to how often someone may use a comma in an env var? Is there a way to escape it via cobra?

**Comment by navidshaikh** [comment]
> but do people have a guess as to how often someone may use a comma in an env var? Is there a way to escape it via cobra?

Supporting this would disallow users from setting commas in env var values, need to check about escaping but single/double quotes on only values or even whole pair, back/forward slash didnt work.

```
➜  client git:(pr/multiple-envs)  ./kn service update svc --env TARGET="hello2,"
--env argument requires a value that contains the '=' character; got 

➜  client git:(pr/multiple-envs)  ./kn service update svc --env TARGET='hello2,'
--env argument requires a value that contains the '=' character; got 

➜  client git:(pr/multiple-envs)  ./kn service update svc --env TARGET=hello2\,  
--env argument requires a value that contains the '=' character; got 

➜  client git:(pr/multiple-envs)  ./kn service update svc --env TARGET=hello2/,
--env argument requires a value that contains the '=' character; got 

➜  client git:(pr/multiple-envs)  ./kn service update svc --env "TARGET=hello2,"
--env argument requires a value that contains the '=' character; got 

# the current behavior allows commas
➜  client git:(pr/multiple-envs)  git checkout master                           
➜  client git:(master)  go run ./cmd/kn/main.go service update svc --env TARGET=hello, 
Waiting for service 'svc' to become ready ... OK
Service 'svc' updated in namespace 'default'.

➜  client git:(master)  curl http://svc.default.apps-crc.testing                      
Hello hello,!
```
The current forma though support it (check last service update command above).

**Comment by duglin** [comment]
I did some digging and found this:
https://github.com/spf13/pflag/pull/59/files

And this does seem to work:
```
./kn service update echo --env=\"abc=def,def=qwe\"
```

So if we document this then I think we're good

**Comment by duglin** [comment]
We should add a test for it too :-)

**Comment by navidshaikh** [comment]
@duglin : Nice! I can do this as well
```
kn service update svc --env \"TARGET=hello,=\",TARGET=overwrite
```

File: pkg/kn/commands/service/service_update_test.go [hunk 1]
```diff
@@ -258,6 +258,65 @@ func TestServiceUpdateEnv(t *testing.T) {
...
+	}
+
+	template, err := servinglib.RevisionTemplateOfService(orig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	servinglib.UpdateImage(template, "gcr.io/foo/bar:baz")
+
+	action, updated, _, err := fakeServiceUpdate(orig, []string{
```
Comment (sixolet) [inline pkg/kn/commands/service/service_update_test.go]: Add test for quoting (to have a comma in an env var), and for repeating the flag, to make sure we cover all the bases.

**Comment by jimcurtis** [comment]
LGTM

**Comment by navidshaikh** [comment]
repeat --env flag to specify multiple env vars, escaping might create more confusion.

**Pull request closed without merging**